packages/babel-plugin-proposal-pipeline-operator/src/index.js

```diff
@@ -15,6 +15,8 @@ export default function() {
 
         let optimizeArrow =
           t.isArrowFunctionExpression(right) && t.isExpression(right.body);
+        const isAwait = t.isAwaitExpression(right);
+        const isYield = t.isYieldExpression(right);
```

Comment:
`foo |> yield bar` shouldn't be transfrmed; it should become something like (`_temp = yield bar, _temp(foo)`).

If there isn't a way of transforming `|> await foo` _before_ transforming the async function, you can workaround the problem by adding `node.__wasAwait` to yield expressions generated by the async functions plugin and then do
```js
const isAwait = t.isAwaitExpression(right) || right.__wasAwait;
```